certgen/certgen_test.go

```diff
@@ -105,9 +126,15 @@ func TestIssueCertificate(t *testing.T) {
 			cc.NotAfterInterval = test.notAfterInterval
 			ca := &CA{SigningCert: root, SigningKey: rootKey, CertConfig: cc}
 
-			cert, err := ca.IssueCertificate()
+			var cert *x509.Certificate
+			var err error
+			if test.precert {
+				cert, err = ca.IssuePrecertificate()
+			} else {
+				cert, err = ca.IssueCertificate()
+			}
 			if err != nil {
-				t.Fatalf("error creating certificate: %s", err)
+				t.Fatalf("error creating (pre)certificate: %s", err)
```

Comment:
If this fails the test you'll get a t.Run(desc) prefix
which hopefully identifies whether it was pre- or not.

But to be safe you might do something like:
`t.Fatalf("error creating (pre? %t)-certificate: %s", test.precert, err)`